server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/QuotaRoutes.java

```diff
@@ -0,0 +1,56 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.webadmin.routes;
+
+import org.apache.james.webadmin.Routes;
+import org.apache.james.webadmin.dto.QuotaDTO;
+import org.apache.james.webadmin.utils.ErrorResponder;
+import org.apache.james.webadmin.utils.JsonExtractException;
+import org.apache.james.webadmin.utils.JsonExtractor;
+import org.eclipse.jetty.http.HttpStatus;
+
+import spark.Request;
+
+public interface QuotaRoutes extends Routes {
+
+    default QuotaDTO parseQuotaDTO(JsonExtractor<QuotaDTO> jsonExtractor, Request request) {
+        try {
+            QuotaDTO dto = jsonExtractor.parse(request.body());
+            if (dto.isValid()) {
+                return dto;
+            } else {
+                throw new IllegalArgumentException();
+            }
+        } catch (IllegalArgumentException e) {
+            throw ErrorResponder.builder()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .type(ErrorResponder.ErrorType.INVALID_ARGUMENT)
+                .message("Invalid quota. Need to be an integer value greater or equal to -1")
+                .cause(e)
+                .haltError();
+        } catch (JsonExtractException e) {
```

Comment:
Should be handled at the webadmin level if I am not wrong.

Reply:
Why ? It's from the `jsonExtractor.parse` method. I prefer it to be handled only in one place.

Reply:
This is handled at the webadmin server level. **For all routes**.

See `WebAdminServer::configureExceptionHanding`:

```
    private void configureExceptionHanding() {
        // ...
        service.exception(JsonExtractException.class, (ex, req, res) -> {
            res.status(BAD_REQUEST_400);
            res.body(ErrorResponder.builder()
                .statusCode(BAD_REQUEST_400)
                .type(INVALID_ARGUMENT)
                .message("JSON payload of the request is not valid")
                .cause(ex)
                .asString());
        });
        // ...
    }
```

Reply:
(the cause should be holding the fact that negative quotas are invalid)

Reply:
Ok, i have added a fixup to throws the `JsonExtractException`. 
Please note that this is not related to this PR as i have just factorized the code and didn't change it's behavior until then.